Update cursor after thought deletion

shresthabijay opened the pull request:
fixes #1455 

## Cause of the issue

After thought deletion the cursor was not updated causing `expandThoughts` function to panic.

## Solution

Update the cursor after thought deletion.

raineorshine reviewed (changes requested):
Thanks! Can you add a unit test?

shresthabijay commented on src/selectors/expandThoughts.ts:
```diff
@@ -69,7 +69,6 @@ function expandThoughts(
     firstVisibleThoughtPath && firstVisibleThoughtPath.length !== 0 ? firstVisibleThoughtPath : HOME_PATH
 
   if (path && !getThoughtById(state, head(path))) {
-    console.error(`expandThought: Base path head thought not found!`)
     return {}
   }
```
This is not expected under normal conditions. This indicated we are calling this function with the wrong path or path that is not available in the tree. So I think we must fail hard.

shresthabijay commented:
@raineorshine I added hard failure if the given path is incorrect in `expandThoughts`. I added proper tests for updating the cursor after thought deletion.

Also, there was this issue with undo-redo after I added hard failure. Turns out it was due to a stale cursor. I have fixed it and  added a comment describing why it is needed. 

https://github.com/cybersemics/em/blob/55d1b817781da3f402baf8a2dad01f5c580b3b33/src/redux-enhancers/undoRedoReducerEnhancer.ts#L37-L47.

I believe this should work now.
